crates/wasmtime/src/compile.rs

```diff
@@ -881,6 +881,22 @@ the use case.
                     }
                 }
             }
+
+            // Skip inlining into array-abi functions which are entry
+            // trampolines into wasm. ABI-wise it's required that these have a
+            // single `try_call` into the module and it doesn't work if multiple
+            // get inlined or if the `try_call` goes away. Pevent all inlining
+            // to guarantee the structure of entry trampolines.
+            (
+                FuncKey::ArrayToWasmTrampoline(..)
+                | FuncKey::ComponentTrampoline(Abi::Array, _)
+                | FuncKey::UnsafeIntrinsic(Abi::Array, _),
+                _,
+            ) => {
+                log::trace!("  --> not inlining: not inlining into array-abi caller");
+                return false;
+            }
```

Comment:
cc @fitzgen on this. One reason is that cranelift inlining doesn't support `get_exception_handler_address` but even if it did we couldn't properly inline in entry trampolines because we wouldn't have the right handler recorded at all call points. To handle that I disabled inlining into these callers no matter what. 

My thinking is that these aren't perf-sensitive and if they are we need to figure something else out other than inlining.